datafusion/functions-nested/src/reverse.rs

```diff
@@ -183,6 +195,72 @@ fn general_array_reverse<O: OffsetSizeTrait + TryFrom<i64>>(
     )?))
 }
 
+fn list_view_reverse<O: OffsetSizeTrait + TryFrom<i64>>(
+    array: &GenericListViewArray<O>,
+    field: &FieldRef,
+) -> Result<ArrayRef> {
+    let offsets = array.offsets();
+    let values = array.values();
+    let sizes = array.sizes();
+
+    // Construct indices, sizes and offsets for the reversed array by iterating over
+    // the list view array in the logical order, and reversing the order of the elements.
+    // We end up with a list view array where the elements are in order,
+    // even if the original array had elements out of order.
+    let mut indices: Vec<O> = Vec::with_capacity(values.len());
+    let mut new_sizes = Vec::with_capacity(sizes.len());
+    let mut new_offsets: Vec<O> = Vec::with_capacity(offsets.len());
+    // Add the offset of the first array
+    new_offsets.push(O::zero());
+    for (i, offset) in offsets.iter().enumerate() {
+        // If this array is null, we set the new array to null with size 0 and continue
+        if array.is_null(i) {
+            new_sizes.push(O::zero());
+            new_offsets.push(new_offsets[i]);
+            continue;
+        }
```

Comment:
Do we have a test case where we have only null inputs? Because if we push an offset at the start, then encounter a ListView with a single null element, I think we end up where offsets length is 2 but sizes length is 1.

Reply:
Great point, this current implementation did indeed fail that test!